### ISSUE TITLE
spam filter tiktok live

### ISSUE THREAD

**Issue by nirvanaruns** [maddox05/Comment-Game-Controller#2]
hey :) if i use :

w = up
s = down
a = left
d = right
z = a button
x = b button
c = start button
v = select button

in visual boy emulator, how can i avoid the spam filter of tiktok lives? is there another type of setup to avoid it like example:

viewer1 : hello streamer nice game playerUp
(python read chat)
(python detect "playerUp")
(python press "w" in keyboard)
(visual boy studio move the character "up" once)

it could be possible like this to avoid the spam filter?

the idea is to get the command in all the ways possibles and not doing like

viewer1 : w
viewer2: w
viewer3: w
viewer1 : w
viewer1 : w

thanks a lot :)

**Comment by maddox05** [comment]
Hi! Do you mean if I typed A sentence with w in it the player would move up? Are you also talking about having more words to move up? Also adding more buttons should be easy for me. For the spam, that could also be protected, but how do you want it to stop spammers? If you answer I can then go and make the changes. Thanks!

**Comment by nirvanaruns** [comment]
hello maddox05 , thanks for answer me :) , I'll try to be more explicit about "the problem"...

It happens that on TikTok if you simply post the letter "w" in a chat, said comment will not be filtered, but if after that you write a "w" again, the letter will not be identified, the streamer will not see it and so will the script. will not do the "raise" action (assuming that with the letter w it does this action). At the same time, I found out that if you write the same thing more than 3 times, nothing identifies you for about 1 minute. As a workaround I modified your code so that instead of using a particular letter for character movement (such as setting the "w" for "go up") it was with a specific word. This is how I managed to arrive at:
If the viewer writes "up" (in Spanish since it is my native language) the script will take said word and bind it to the "w" key.
Finally I also made the latter be within a sentence for example:
viewer1 writes in the chat: hello stream up how are you?
the script will identify that "up" and do the action.
At this moment I am not at home to send you a copy of the code but later if you arrive I will send it to you here if you want.

Regarding the issue of tiktok spam, I did not find another way to avoid said block. I forgot that I also found that lives that use tikfinity also have a considerable block on the script connection to said channels. (no idea why but I imagine tikfinity has a pretty powerful anti-spam filter).
